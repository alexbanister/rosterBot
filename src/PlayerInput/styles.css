.playerInput .form-block {
  border: 1px solid #3ee33e; }

.playerInput input {
  display: inline-block;
  margin: 1em auto; }

.playerInput .attr {
  width: 4em; }

.playerInput .error {
  color: #c24242; }
