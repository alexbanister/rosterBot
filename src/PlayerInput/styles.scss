@import "../variables.scss";

.playerInput {
  border: 2px solid #3ee33e;
  input {
    margin: .25em;
  }
  .attr {
    width: 4em;
  }
  .errorMessage {
    width: 2em;
    visibility: hidden;
    img {
      width: 1.5em;
    }
  }
}
.error {
  input {
    background: $error;
  }
  .tooltip {
    visibility: visible;
    cursor: pointer;
    position: relative;
    display: inline-block;
  }
  .tooltip .tooltiptext {
      visibility: hidden;
      width: 25em;
      top: 100%;
      left: 50%;
      margin-left: -23.5em;
      background-color: $body-dark;
      color: $body-light;
      text-align: center;
      padding: 1em;
      border-radius: .5em;
      position: absolute;
      z-index: 1;
      box-shadow: 5px 5px 5px $background-dark;
  }
  .tooltip:hover .tooltiptext {
      visibility: visible;
  }
}
