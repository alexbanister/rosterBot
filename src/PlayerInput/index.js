import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './styles.css';

export class PlayerInput extends Component {
  constructor({...props}) {
    super();
    this.state = {
      firstName: props.firstName || '',
      lastName: props.lastName || '',
      speed: props.speed || 0,
      strength: props.strength || 0,
      agility: props.agility || 0,
      role: props.playerRole
    };
  }

  handleChange(field, value){
    this.setState({
      [field]: value
    });
  }

  keepAsNumber(num) {
    num = num ? num : 0;
    return parseInt(num, 10);
  }

  getStatTotal({ speed, agility, strength }){
    return speed + agility + strength;
  }

  displayErrorMessages(messages) {
    return messages.map((message, iter) => {
      return <li key={iter}>{message}</li>;
    });
  }

  render() {
    const isError = this.props.error ? 'error' : 'clean';
    return (
      <tr className={`playerInput ${isError}`}>
        <td>
          <input
            type='text'
            name={`${this.props.name}FirstName`}
            placeholder='First Name'
            value={this.state.firstName}
            onChange={(event) => this.handleChange('firstName', event.target.value)}
            onBlur={() => this.props.savePlayer(this.state)}
          />
        </td>
        <td>
          <input
            type='text'
            name={`${this.props.name}LastName`}
            placeholder='Last Name'
            value={this.state.lastName}
            onChange={(event) => this.handleChange('lastName', event.target.value)}
            onBlur={() => {
              this.props.savePlayer(this.state);
              this.props.validatePlayer(this.state);
            }}
          />
        </td>
        <td>
          <input
            type='number'
            name={`${this.props.name}Speed`}
            className='attr'
            value={this.state.speed}
            onChange={(event) => this.handleChange('speed', this.keepAsNumber(event.target.value, 10))}
            onBlur={() => {
              this.props.savePlayer(this.state);
              this.props.validatePlayer(this.state);
            }}
          />
        </td>
        <td>
          <input
            type='number'
            name={`${this.props.name}Strength`}
            className='attr'
            value={this.state.strength}
            onChange={(event) => this.handleChange('strength', this.keepAsNumber(event.target.value, 10))}
            onBlur={() => {
              this.props.savePlayer(this.state);
              this.props.validatePlayer(this.state);
            }}
          />
        </td>
        <td>
          <input
            type='number'
            name={`${this.props.name}Agility`}
            className='attr'
            value={this.state.agility}
            onChange={(event) => this.handleChange('agility', this.keepAsNumber(event.target.value, 10))}
            onBlur={() => {
              this.props.savePlayer(this.state);
              this.props.validatePlayer(this.state);
            }}
          />
        </td>
        <td>
          { this.getStatTotal(this.state) }
        </td>
        <td className='errorMessage'>
          <div className='tooltip'>
            <img src={process.env.PUBLIC_URL + '/warning.svg'} alt='Error Icon'/>
            <span className='tooltiptext'>
              <h5>Errors!</h5>
              <ul>
                {this.displayErrorMessages(this.props.errorMessage)}
              </ul>
            </span>
          </div>
        </td>
      </tr>
    );
  }
}

PlayerInput.propTypes ={
  name: PropTypes.string,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  speed: PropTypes.number,
  strength: PropTypes.number,
  agility: PropTypes.number,
  playerRole: PropTypes.string,
  errorMessage: PropTypes.array,
  error: PropTypes.bool,
  savePlayer: PropTypes.func,
  validatePlayer: PropTypes.func
};

export default PlayerInput;
