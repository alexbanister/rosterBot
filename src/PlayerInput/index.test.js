import { shallow } from 'enzyme';
import PlayerInput from './index';
import React from 'react';


describe('PlayerInput', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<PlayerInput errorMessage={[]} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should update state', () => {
    const initialState = {
      firstName: '',
      lastName: '',
      speed: 0,
      strength: 0,
      agility: 0,
      role: 'starter'
    };
    const expectation = {
      firstName: 'R2',
      lastName: 'D2',
      speed: 12,
      strength: 49,
      agility: 31,
      role: 'starter'
    };
    const wrapper = shallow(<PlayerInput playerRole='starter' name='starter' errorMessage={[]} />);

    expect(wrapper.state()).toEqual(initialState);

    const firstName = wrapper.find('[name="starterFirstName"]').find('input');
    const lastName = wrapper.find('[name="starterLastName"]').find('input');
    const speed = wrapper.find('[name="starterSpeed"]').find('input');
    const agility = wrapper.find('[name="starterAgility"]').find('input');
    const strength = wrapper.find('[name="starterStrength"]').find('input');
    // console.log(wrapper.debug());
    // console.log(firstName);

    firstName.simulate('change', { target: { value: 'R2' }});
    lastName.simulate('change', { target: { value: 'D2' }});
    speed.simulate('change', { target: { value: '12' }});
    agility.simulate('change', { target: { value: '31' }});
    strength.simulate('change', { target: { value: '49' }});

    expect(wrapper.state()).toEqual(expectation);
  });
});
