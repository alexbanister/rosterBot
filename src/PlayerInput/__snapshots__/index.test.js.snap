// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerInput should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerInput
    errorMessage={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          <input
                    name="undefinedFirstName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="First Name"
                    type="text"
                    value=""
          />
</td>,
        <td>
          <input
                    name="undefinedLastName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Last Name"
                    type="text"
                    value=""
          />
</td>,
        <td>
          <input
                    className="attr"
                    name="undefinedSpeed"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="number"
                    value={0}
          />
</td>,
        <td>
          <input
                    className="attr"
                    name="undefinedStrength"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="number"
                    value={0}
          />
</td>,
        <td>
          <input
                    className="attr"
                    name="undefinedAgility"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="number"
                    value={0}
          />
</td>,
        <td />,
        <td>
          <div
                    className="errorMessage"
          >
                    <div
                              className="tooltip"
                    >
                              <img
                                        alt="Error Icon"
                                        src="/warning.svg"
                              />
                              <span
                                        className="tooltiptext"
                              >
                                        <h5>
                                                  Errors!
                                        </h5>
                                        <ul>
                                                  
                                        </ul>
                              </span>
                    </div>
          </div>
</td>,
      ],
      "className": "playerInput clean",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="undefinedFirstName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="First Name"
            type="text"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "undefinedFirstName",
            "onBlur": [Function],
            "onChange": [Function],
            "placeholder": "First Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="undefinedLastName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "undefinedLastName",
            "onBlur": [Function],
            "onChange": [Function],
            "placeholder": "Last Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="attr"
            name="undefinedSpeed"
            onBlur={[Function]}
            onChange={[Function]}
            type="number"
            value={0}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "attr",
            "name": "undefinedSpeed",
            "onBlur": [Function],
            "onChange": [Function],
            "type": "number",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="attr"
            name="undefinedStrength"
            onBlur={[Function]}
            onChange={[Function]}
            type="number"
            value={0}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "attr",
            "name": "undefinedStrength",
            "onBlur": [Function],
            "onChange": [Function],
            "type": "number",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="attr"
            name="undefinedAgility"
            onBlur={[Function]}
            onChange={[Function]}
            type="number"
            value={0}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "attr",
            "name": "undefinedAgility",
            "onBlur": [Function],
            "onChange": [Function],
            "type": "number",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
        },
        "ref": null,
        "rendered": 0,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="errorMessage"
>
            <div
                        className="tooltip"
            >
                        <img
                                    alt="Error Icon"
                                    src="/warning.svg"
                        />
                        <span
                                    className="tooltiptext"
                        >
                                    <h5>
                                                Errors!
                                    </h5>
                                    <ul>
                                                
                                    </ul>
                        </span>
            </div>
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="tooltip"
>
              <img
                            alt="Error Icon"
                            src="/warning.svg"
              />
              <span
                            className="tooltiptext"
              >
                            <h5>
                                          Errors!
                            </h5>
                            <ul>
                                          
                            </ul>
              </span>
</div>,
            "className": "errorMessage",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Error Icon"
                  src="/warning.svg"
/>,
                <span
                  className="tooltiptext"
>
                  <h5>
                                    Errors!
                  </h5>
                  <ul>
                                    
                  </ul>
</span>,
              ],
              "className": "tooltip",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Error Icon",
                  "src": "/warning.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h5>
                      Errors!
</h5>,
                    <ul>
                      
</ul>,
                  ],
                  "className": "tooltiptext",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Errors!",
                    },
                    "ref": null,
                    "rendered": "Errors!",
                    "type": "h5",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [],
                    },
                    "ref": null,
                    "rendered": Array [],
                    "type": "ul",
                  },
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            <input
                        name="undefinedFirstName"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="First Name"
                        type="text"
                        value=""
            />
</td>,
          <td>
            <input
                        name="undefinedLastName"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Last Name"
                        type="text"
                        value=""
            />
</td>,
          <td>
            <input
                        className="attr"
                        name="undefinedSpeed"
                        onBlur={[Function]}
                        onChange={[Function]}
                        type="number"
                        value={0}
            />
</td>,
          <td>
            <input
                        className="attr"
                        name="undefinedStrength"
                        onBlur={[Function]}
                        onChange={[Function]}
                        type="number"
                        value={0}
            />
</td>,
          <td>
            <input
                        className="attr"
                        name="undefinedAgility"
                        onBlur={[Function]}
                        onChange={[Function]}
                        type="number"
                        value={0}
            />
</td>,
          <td />,
          <td>
            <div
                        className="errorMessage"
            >
                        <div
                                    className="tooltip"
                        >
                                    <img
                                                alt="Error Icon"
                                                src="/warning.svg"
                                    />
                                    <span
                                                className="tooltiptext"
                                    >
                                                <h5>
                                                            Errors!
                                                </h5>
                                                <ul>
                                                            
                                                </ul>
                                    </span>
                        </div>
            </div>
</td>,
        ],
        "className": "playerInput clean",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="undefinedFirstName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="First Name"
              type="text"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "undefinedFirstName",
              "onBlur": [Function],
              "onChange": [Function],
              "placeholder": "First Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="undefinedLastName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Last Name"
              type="text"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "undefinedLastName",
              "onBlur": [Function],
              "onChange": [Function],
              "placeholder": "Last Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="attr"
              name="undefinedSpeed"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "attr",
              "name": "undefinedSpeed",
              "onBlur": [Function],
              "onChange": [Function],
              "type": "number",
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="attr"
              name="undefinedStrength"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "attr",
              "name": "undefinedStrength",
              "onBlur": [Function],
              "onChange": [Function],
              "type": "number",
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="attr"
              name="undefinedAgility"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "attr",
              "name": "undefinedAgility",
              "onBlur": [Function],
              "onChange": [Function],
              "type": "number",
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="errorMessage"
>
              <div
                            className="tooltip"
              >
                            <img
                                          alt="Error Icon"
                                          src="/warning.svg"
                            />
                            <span
                                          className="tooltiptext"
                            >
                                          <h5>
                                                        Errors!
                                          </h5>
                                          <ul>
                                                        
                                          </ul>
                            </span>
              </div>
</div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="tooltip"
>
                <img
                                alt="Error Icon"
                                src="/warning.svg"
                />
                <span
                                className="tooltiptext"
                >
                                <h5>
                                                Errors!
                                </h5>
                                <ul>
                                                
                                </ul>
                </span>
</div>,
              "className": "errorMessage",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="Error Icon"
                    src="/warning.svg"
/>,
                  <span
                    className="tooltiptext"
>
                    <h5>
                                        Errors!
                    </h5>
                    <ul>
                                        
                    </ul>
</span>,
                ],
                "className": "tooltip",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Error Icon",
                    "src": "/warning.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h5>
                        Errors!
</h5>,
                      <ul>
                        
</ul>,
                    ],
                    "className": "tooltiptext",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Errors!",
                      },
                      "ref": null,
                      "rendered": "Errors!",
                      "type": "h5",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [],
                      },
                      "ref": null,
                      "rendered": Array [],
                      "type": "ul",
                    },
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
